TestWebKitAPIApp fails to build with error: TestWebKitAPIResources.bundle: No such file or directory

aestes opened the pull request:
#### 1057d5aa1b8b9058c73e46ba2ccd407184904135
<pre>
TestWebKitAPIApp fails to build with error: TestWebKitAPIResources.bundle: No such file or directory
<a href="https://bugs.webkit.org/show_bug.cgi?id=288138">https://bugs.webkit.org/show_bug.cgi?id=288138</a>

Reviewed by Abrar Rahman Protyasha and Elliott Williams.

When building a scheme that enables &quot;Find Implicit Dependencies&quot; and includes TestWebKitAPIApp,
the build fails with errors such as:

    error: /.../WebKitBuild/Debug-iphoneos/TestWebKitAPIResources.bundle: No such file or directory
    (in target &apos;TestWebKitAPIApp&apos; from project &apos;TestWebKitAPI&apos;)

TestWebKitAPIResources.bundle was included in TestWebKitAPIApp&apos;s &quot;Copy Bundle Resources&quot; build
phase, but it turns out that this type of build phase does not establish implicit dependencies.
Replaced it with a &quot;Copy Files&quot; build phase, which does establish implicit dependencies.

* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/290769@main">https://commits.webkit.org/290769@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1ece759f77a7f6ca52e63b4c51f0cda8c26af5e2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91049 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/10592 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/60 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/96068 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/41834 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/93101 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/10987 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/18910 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/69982 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/27510 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94050 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/8388 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/82519 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50308 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/8160 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/47 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/40958 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/78487 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/45 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98045 "") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/18247 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/13401 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/98045 "") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/18506 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/78327 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/98045 "") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19322 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/22688 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/57 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/11490 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/18253 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/17996 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/21462 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/19780 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
```diff
@@ -1156,16 +1156,16 @@
 		A18899012CA9FA6700FEB09A /* Cocoa.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = BCB9E9F011235BDE00A137E0 /* Cocoa.framework */; platformFilters = (macos, ); };
 		A198D0A22BC64D8E0058BBEB /* NowPlayingMetadataObserver.mm in Sources */ = {isa = PBXBuildFile; fileRef = A198D0A12BC64D8E0058BBEB /* NowPlayingMetadataObserver.mm */; };
 		A1AD382F2C3DA40B003499BE /* FullscreenLifecycle.mm in Sources */ = {isa = PBXBuildFile; fileRef = A1AD38272C3D9FAF003499BE /* FullscreenLifecycle.mm */; };
+		A1B8D76A2D67CD830045C305 /* InjectedBundleTestWebKitAPI.bundle in Copy Resources */ = {isa = PBXBuildFile; fileRef = BC575980126E74AF006F0F12 /* InjectedBundleTestWebKitAPI.bundle */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
```
Are these attributes necessary?

aestes replied:
AFAICT there is no UI setting to control `RemoveHeadersOnCopy`, so while it doesn't seem necessary I don't know how I would remove it.

`CodeSignOnCopy` may not actually be necessary; I'll test and see.

emw-apple replied:
I wouldn't be shocked if CodeSignOnCopy _is_ necessary, at least in internal builds where we intend for TestWebKitAPI to be redistributable. Thank you for testing!

aestes replied:
True. I guess I don't know the difference between code-signing the entire TestWebKitAPI.app bundle (which occurs regardless of this attribute) and code-signing these individual resources on copy.

aestes replied:
Since two of these bundles contain executable code that we load at runtime, and we intend to run TestWebKitAPI.app on iOS devices, it appears that code-signing on copy is necessary (and at the very least isn't harmful), so I'll keep the attribute.

aprotyas reviewed (approved):
LGTM with a question

aestes committed:
TestWebKitAPIApp fails to build with error: TestWebKitAPIResources.bundle: No such file or directory
https://bugs.webkit.org/show_bug.cgi?id=288138

Reviewed by Abrar Rahman Protyasha and Elliott Williams.

When building a scheme that enables "Find Implicit Dependencies" and includes TestWebKitAPIApp,
the build fails with errors such as:

    error: /.../WebKitBuild/Debug-iphoneos/TestWebKitAPIResources.bundle: No such file or directory
    (in target 'TestWebKitAPIApp' from project 'TestWebKitAPI')

TestWebKitAPIResources.bundle was included in TestWebKitAPIApp's "Copy Bundle Resources" build
phase, but it turns out that this type of build phase does not establish implicit dependencies.
Replaced it with a "Copy Files" build phase, which does establish implicit dependencies.

* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/290769@main
